test_setup.js

```diff
@@ -151,24 +152,87 @@ function assertGen3Client() {
-  bash.runCommand(fenceCmd, 'fence');
-
-  bucketId = fenceProps.googleBucketInfo.test.bucketId
-  googleProjectId = fenceProps.googleBucketInfo.test.googleProjectId
-  projectAuthId = 'test';
-  fenceCmd = `fence-create google-bucket-create --unique-name ${bucketId} --google-project-id ${googleProjectId} --project-auth-id ${projectAuthId} --public False`;
-  console.log(`Running: ${fenceCmd}`);
-  response = bash.runCommand(fenceCmd, 'fence');
-
-  console.log('Clean up Google Bucket Access Groups from previous runs...');
-  bash.runJob('google-verify-bucket-access-group');
+  try {
+    console.log('Ensure test buckets are linked to projects in this commons...');
+
+    var bucketId = fenceProps.googleBucketInfo.QA.bucketId;
+    var googleProjectId = fenceProps.googleBucketInfo.QA.googleProjectId;
+    var projectAuthId = 'QA';
+    var fenceCmd = `fence-create google-bucket-create --unique-name ${bucketId} --google-project-id ${googleProjectId} --project-auth-id ${projectAuthId} --public False`;
+    console.log(`Running: ${fenceCmd}`);
+    bash.runCommand(fenceCmd, 'fence');
+
+    bucketId = fenceProps.googleBucketInfo.test.bucketId
+    googleProjectId = fenceProps.googleBucketInfo.test.googleProjectId
+    projectAuthId = 'test';
+    fenceCmd = `fence-create google-bucket-create --unique-name ${bucketId} --google-project-id ${googleProjectId} --project-auth-id ${projectAuthId} --public False`;
+    console.log(`Running: ${fenceCmd}`);
+    response = bash.runCommand(fenceCmd, 'fence');
+
+    console.log('Clean up Google Bucket Access Groups from previous runs...');
+    bash.runJob('google-verify-bucket-access-group');
+  }
+  catch (e) {
+    if (inJenkins) {
+      throw e;
+    }
+    console.log('WARNING: unable to create Google test buckets. You can ignore this message if you do not want to run Google data access tests.');
+  }
+}
+
+async function setupGoogleProjectDynamic() {
+  // Update the id and SA email depending on the current namespace
+  if (process.env.RUNNING_LOCAL) { // local run
+    namespace = 'validationjobtest';
+  }
+  else { // jenkins run. a google project exists for each jenkins env
+    namespace = process.env.NAMESPACE;
+  }
+  fenceProps.googleProjectDynamic.id = fenceProps.googleProjectDynamic.id.replace(
+    'NAMESPACE',
+    namespace
+  );
+  fenceProps.googleProjectDynamic.serviceAccountEmail = fenceProps.googleProjectDynamic.serviceAccountEmail.replace(
+    'NAMESPACE',
+    namespace
+  );
+  console.log(`googleProjectDynamic: ${fenceProps.googleProjectDynamic.id}`);
+
+  // Add the IAM access needed by the monitor service account
+  const monitorRoles = [
+    'roles/resourcemanager.projectIamAdmin',
+    'roles/editor'
+  ];
+  for (var role of monitorRoles) {
+    let res = await google.updateUserRole(
+      fenceProps.googleProjectDynamic.id,
+      {
+        role,
+        members: [`serviceAccount:${fenceProps.monitorServiceAccount}`]
+      }
+    );
+    if (res.code) {
```

Comment:
shouldn't this check be something like `if res.code != 200`?

Reply:
if it works there's no `code` at all 🤷‍♀️

Reply:
:confounded: welp, makes sense then